### ISSUE TITLE
pivy-9999: swig fails to find c headers

### ISSUE THREAD

**Issue by alexanderustinov** [waebbl/waebbl-gentoo#39]
Greetings again :)
Now pivy's setup.py fails.

Error messages like
`Inventor/SbDict.h:53: Error: Unable to find 'cstddef'`
`Inventor/SbString.h:40: Error: Unable to find 'cstdarg'`
`Inventor/lists/SbPList.h:41: Error: Unable to find 'cassert'`
`Inventor/lists/SbPList.h:42: Error: Unable to find 'cstddef'`
`/usr/include/Inventor/lists/SbList.h:36: Error: Unable to find 'cassert'`
`/usr/include/Inventor/lists/SbList.h:37: Error: Unable to find 'cstddef'`
are triggered by `swig -w302,306,307,312,314,325,361,362,467,389,503,509,510 -py3 -c++ -python -includeall -modern -D__PIVY__ -I. -Ifake_headers -I"/usr/include" -Iinterfaces  -o pivy/coin_wrap.cpp interfaces/coin.i`.

Named headers are available at /usr/include/boost/compatibility/cpp_c_headers/ , which swig fails to recurse into.

Can additional include directories be supplied by ebuild file to the build script?

**Comment by waebbl** [comment]
I works on my machine using coin-4.0.0a_pre20180921 and SoQt-1.6.0a_pre20180813.
I'd prefer to loosen the dependencies and only depend on coin and SoQt, but not necesserily a live version. What do you think?

**Comment by alexanderustinov** [comment]
Where do you have cstddef file(s) inside /usr/include? 
Would adding a dependency on sys-libs/libcxx be a better resolution?

**Comment by waebbl** [comment]
These are the locations where I have `cstddef`
```
/usr/include/boost/compatibility/cpp_c_headers/cstddef
/usr/include/c++/v1/cstddef
/usr/lib64/gcc/x86_64-pc-linux-gnu/7.3.0/include/g++-v7/cstddef
```
They belong to boost, libcxx and gcc. I don't think we have to add either of boost or libstdcxx. Gcc comes  packaged with the needed headers, we just need to figure out, why the build is not finding those files.

**Comment by alexanderustinov** [comment]
Touching those files seems like way to go.

I have swig 3.0.12, is yours version number different?

**Comment by waebbl** [comment]
No, I'm using swig-3.0.12 too.
What do you think, is it an upstream issue or a gentoo specific one?

**Comment by alexanderustinov** [comment]
It is both, i suppose.

To track this down we need to understand  why it does not work for me and why it works for you.
What will somewhat `like su - -s /bin/bash portage -c 'cd /var/tmp/portage/dev-python/pivy-9999/work/pivy && swig -w302,306,307,312,314,325,361,362,467,389,503,509,510 -py3 -c++ -python -includeall -modern -D__PIVY__ -I. -Ifake_headers -I"/usr/include" -Iinterfaces  -o pivy/coin_wrap.cpp interfaces/coin.i'` give on your machine?
Does the call i mention resemble the call portage makes when it emerges pivy? (for me it comes shortly after 'Preparing Inventor headers')

**Comment by waebbl** [comment]
I tried to `ebuild pivy-9999.ebuild unpack` as normal user and then run your command like `/bin/bash -c [...]` and it succeeded in generating coin_wrap.cpp. I got a lot of swig warnings on overloaded methods which are shadowed, but it built the C++ file.

But... the last time I installed coin and SoQt was on October 3rd, using the already mentioned versions, so on my machine, those versios have been there for quite some time. When I tried to install pivy-9999 in a 64bit clean chroot yesterday, I got the same errors like you, but the `touch` patch fixed this.

I checked coin at bitbucket today and found they had quite some updates for coin and SoQt, SoGui. Looks like I need to update my ebuilds for coin and SoQt.

I remember, when we first set up the triple coin, soqt and pivy we also had a lot of trouble finding a combination of commits which did work together properly. I suppose upstream doesn't have a good CI which also checks library interactions when something is updated.

Additionally someone posted a bug with pivy-0.6.4 today at freecad forum: https://forum.freecadweb.org/viewtopic.php?f=4&t=32831
